jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cwc/auth/AuthControllerTest.kt

```diff
@@ -0,0 +1,185 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cwc.auth
+
+import com.intellij.openapi.project.Project
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.mockkStatic
+import io.mockk.unmockkAll
+import io.mockk.verify
+import kotlinx.coroutines.runBlocking
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import software.aws.toolkits.jetbrains.core.gettingstarted.editor.ActiveConnection
+import software.aws.toolkits.jetbrains.core.gettingstarted.editor.BearerTokenFeatureSet
+import software.aws.toolkits.jetbrains.core.gettingstarted.editor.checkBearerConnectionValidity
+import software.aws.toolkits.jetbrains.core.gettingstarted.reauthenticateWithQ
+import software.aws.toolkits.jetbrains.services.cwc.controller.chat.telemetry.TelemetryHelper
+import software.aws.toolkits.jetbrains.services.cwc.utility.EdtUtility
+import software.aws.toolkits.telemetry.CwsprChatCommandType
+
+class AuthControllerTest {
+
+    private val mockProject: Project = mockk<Project>(relaxed = true)
+
+    @Before
+    fun setUp() {
+        mockkStatic("software.aws.toolkits.jetbrains.core.gettingstarted.editor.GettingStartedPanelUtilsKt")
+    }
+
+    @After
+    fun tearDown() {
+        unmockkAll()
+    }
+
+    @Test
+    fun `NotConnected returns AuthNeededState`() {
+        // Arrange
+        every { checkBearerConnectionValidity(any(), BearerTokenFeatureSet.Q) } returns ActiveConnection.NotConnected
```

Comment:
You can make a helper function to avoid repeating all of these calls.

```kotlin
private fun mockConnectionValidity(
  qConnection: ActiveConnection = ActiveConnection.NotConnected,
  cwConnection: ActiveConnection = ActiveConnection.NotConnected
) {
    every { checkBearerConnectionValidity(any(), BearerTokenFeatureSet.Q) } returns qConnection
    every { checkBearerConnectionValidity(any(), BearerTokenFeatureSet.CODEWHISPERER) } returns cwConnection
}
```